features/cloud.py

```diff
@@ -0,0 +1,373 @@
+import os
+import logging
+import pycloudlib  # type: ignore
+import time
+import yaml
+
+from typing import Optional
+
+
+class Cloud:
+    """Base class for cloud providers that should be tested through behave.
+
+    :param tag:
+        A tag to be used when creating the resources on the cloud provider
+    """
+
+    name = ""
+
+    def __init__(self, tag):
+        self.tag = tag
+        self._api = None
+
+    def has_keys(self):
+        """Check if cloud has necessary keys to launch instance."""
+        raise NotImplementedError
+
+    def get_instance_id(self, instance):
+        """Return the instance identifier.
+
+        :param instance:
+            An instance created on the cloud provider
```

Comment:
The id of an instance created in the cloud.
```suggestion
            The string of the unique instance id.
```